src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -512,27 +418,32 @@ public void setVectorZ(final int vectorZ)
      */
     public BlockPos getCobbleLocation()
     {
+        if (cobbleLocation == null)
+        {
+            loadLadderPos();
+        }
+
         return cobbleLocation;
     }
 
-    /**
-     * Setter for the cobbleLocation.
-     *
-     * @param pos the location to set.
-     */
-    public void setCobbleLocation(final BlockPos pos)
+    @Override
+    public void setTileEntity(final AbstractTileEntityColonyBuilding te)
     {
-        this.cobbleLocation = pos;
+        super.setTileEntity(te);
+        loadLadderPos();
     }
 
-    /**
-     * Setter of the shaftStart.
-     *
-     * @param pos the location.
-     */
-    public void setShaftStart(final BlockPos pos)
+    private void loadLadderPos()
     {
-        this.shaftStart = pos;
+        final Set<BlockPos> cobblePos = tileEntity.getWorldTagNamePosMap().get("cobble");
```

Comment:
isnt it easier to keep the map var and check on that? also you'll get null values when it doesnt exist, its a hashmap